### PR TITLE
Auto Evo Config is stored in JSON

### LINKED ISSUE #1020: Make auto-evo parameters load from JSON
https://github.com/Revolutionary-Games/Thrive/blob/bf37ff573abcdafe05017db82999c9669449d83c/src/auto-evo/AutoEvoRun.cs#L14

### PR DESCRIPTION
**Pull request by icedjuro**
Closes #1020

### CODE REVIEW

File: src/auto-evo/AutoEvoConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+public class AutoEvoConfiguration : IRegistryType
+{
+    public int MUTATIONS_PER_SPECIES;
+    public bool ALLOW_NO_MUTATION;
+    public int MOVE_ATTEMPTS_PER_SPECIES;
+    public bool ALLOW_NO_MIGRATION;
```
Comment (hhyyrylainen) [inline src/auto-evo/AutoEvoConfiguration.cs:6]: These should be named according to the normal C# field. Also converting these to properties with private set and the jsonproperty attribute would protect against someone trying to modify these during runtime.

File: src/auto-evo/AutoEvoConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    /// Unused
+    /// </summary>
+    /// <value>The name of the internal.</value>
+    public string InternalName { get; set; }
+
+    public void Check(string name)
+    {
+        if (MUTATIONS_PER_SPECIES < 0)
+        {
+            throw new InvalidRegistryDataException("NameGenerator", GetType().Name,
```
Comment (hhyyrylainen) [inline src/auto-evo/AutoEvoConfiguration.cs:18]: `NameGenerator` doesn't belong here

File: simulation_parameters/common/autoevo_parameters.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "MUTATIONS_PER_SPECIES": 3,
+    "ALLOW_NO_MUTATION": true,
+    "MOVE_ATTEMPTS_PER_SPECIES": 5,
+    "ALLOW_NO_MIGRATION": true
```
Comment (hhyyrylainen) [inline simulation_parameters/common/autoevo_parameters.json:5]: The new naming needs to also apply here

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/auto-evo/AutoEvoConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    [JsonProperty]
+    public int MoveAttemptsPerSpecies { get; private set; }
+
+    [JsonProperty]
+    public bool AllowNoMigration { get; private set; }
+
+    /// <summary>
+    /// Unused
+    /// </summary>
+    /// <value>The name of the internal.</value>
```
Comment (hhyyrylainen) [inline src/auto-evo/AutoEvoConfiguration.cs:20]: What even is this comment. Unused is fine to keep but it should be intended 2 spaces.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

**Review by hhyyrylainen — APPROVED** [review]

**Comment by hhyyrylainen** [comment]
You probably need to add a BOM now to that one new file you added for CI to accept it.

**Comment by icedjuro** [comment]
@hhyyrylainen how do i do that?

**Comment by hhyyrylainen** [comment]
Google "YOUREDITOR UTF8 BOM" to see how you can save files with BOM.

**Pull request merged**